src/services/WolfDen.Application/Requests/Queries/Attendence/AttendanceSummary/AttendanceSummaryQueryHandler.cs

```diff
@@ -0,0 +1,131 @@
+        {
+            int minWorkDuration = 360;
+
+            DateOnly monthStart = new DateOnly(request.Year, request.Month, 1);
+            DateOnly monthEnd = monthStart.AddMonths(1).AddDays(-1);
+
+            AttendanceSummaryDTO summaryDto = new AttendanceSummaryDTO
+            {
+                Present = 0,
+                Absent = 0,
+                IncompleteShift = 0,
+                RestrictedHoliday = 0,
+                NormalHoliday = 0,
+                WFH = 0,
+                Leave = 0
+            };
+
+            DateOnly today = DateOnly.FromDateTime(DateTime.Now);
+
+            List<DailyAttendence> attendanceRecords = await _context.DailyAttendence
+                .Where(x => x.EmployeeId == request.EmployeeId && x.Date >= monthStart && x.Date <= monthEnd)
+                .ToListAsync(cancellationToken);
+
+            List<Holiday> holidays = await _context.Holiday
+                .Where(x => x.Date >= monthStart && x.Date <= monthEnd)
+                .ToListAsync(cancellationToken);
+
+            List<LeaveRequest> leaveRequests = await _context.LeaveRequests
+                .Where(x => x.EmployeeId == request.EmployeeId &&
+                            x.FromDate <= monthEnd && x.ToDate >= monthStart &&
+                            x.LeaveRequestStatusId == LeaveRequestStatus.Approved)
+                .ToListAsync(cancellationToken);
+
+            List<LeaveType> leaveTypes = await _context.LeaveType.ToListAsync(cancellationToken);
+
+
+            for (var currentDate = monthStart; currentDate <= monthEnd; currentDate = currentDate.AddDays(1))
+            {
+                if (currentDate > today)
+                {
+                    continue;
+                }
+
+                DailyAttendence attendanceRecord = attendanceRecords.FirstOrDefault(x => x.Date == currentDate);
+                if (attendanceRecord is not null)
+                {
+
+                    if (attendanceRecord.InsideDuration >= minWorkDuration)
+                    {
+                        summaryDto.Present++;
+                    }
+                    else
+                    {
+                        summaryDto.IncompleteShift++;
+                    }
+                }
+                else
+                {
+                    Holiday holiday = holidays.FirstOrDefault(x => x.Date == currentDate);
+                    if (holiday is not null)
+                    {
+
+                        if (holiday.Type is AttendanceStatus.NormalHoliday)
+                        {
+                            summaryDto.NormalHoliday++;
+                        }
+
+                        else if (holiday.Type is AttendanceStatus.RestrictedHoliday)
+                        {
+                            LeaveRequest leaveRequestForHoliday = leaveRequests.FirstOrDefault(x => x.FromDate <= currentDate && x.ToDate >= currentDate);
```

Comment:
why <= and >=, can we use ==

Reply:
in leave request table, duration of leave is available (from date-to date)
We are checking whether current date falls under that duration